File: examples/thrift-server/thrift-example-server/src/test/scala/com/twitter/calculator/CalculatorServerStartupTest.scala
package com.twitter.calculator

import com.google.inject.Stage
import com.twitter.finatra.thrift.EmbeddedThriftServer
import com.twitter.inject.server.FeatureTest

class CalculatorServerStartupTest extends FeatureTest {

  val server =
    new EmbeddedThriftServer(twitterServer = new CalculatorServer, stage = Stage.PRODUCTION)

  test("server#startup") {
    server.assertHealthy()
  }
}

File: examples/java-thrift-server/thrift-example-server/src/test/java/com/twitter/calculator/CalculatorServerStartupTest.java
package com.twitter.calculator;

import com.google.common.collect.ImmutableMap;
import com.google.inject.Stage;

import org.junit.Test;

import com.twitter.finatra.thrift.EmbeddedThriftServer;

public class CalculatorServerStartupTest {

    private EmbeddedThriftServer server =
        new EmbeddedThriftServer(
            new CalculatorServer(),
            ImmutableMap.of(),
            Stage.PRODUCTION);

    /** Test server startup */
    @Test
    public void testServerStartup() {
        server.assertHealthy(true);
        server.close();
    }
}
